### ISSUE TITLE
[Bug]: ports connects to wrong side of pipe

### ISSUE THREAD

**Issue by ladderup** [Project-OMOTES/user_feedback#61]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### What happened?

![image](https://github.com/user-attachments/assets/b0efde88-ce89-43aa-9192-ce43901d522a)

NWN editor normal
Map editor wrong

### Steps to reproduce the problem

NWN editor normal
Map editor wrong

edsl attached (.esdl file)
[dynamicenviromentcopy-20241010 unRun - Copy.txt](https://github.com/user-attachments/files/17327967/dynamicenviromentcopy-20241010.unRun.-.Copy.txt)

### What should have happened?

x,y coordinates fix
but it is not critical bug.

### What browsers do you use to access the UI?

None

**Comment by ewoudwerkman** [comment]
Hi @ladderup, nice one! Did you draw this network in the TPG frontend?
If so, we need to check with @WouterSpaak to find out how the ordering of ports of an asset is handled. According to the ESDL spec of a [Pipe](https://energytransition.github.io/#router/doc-content/687474703a2f2f7777772e746e6f2e6e6c2f6573646c/Pipe.html), the location of the first port correspondences with first point of the line geometry. 
I'm wondering how the TPG frontend is determining the connections to ports on the map?

**Comment by WouterSpaak** [comment]
Looks like this happens when you export ESDL from our frontend and import it into MapEditor.

Given this screenshot:

![image](https://github.com/user-attachments/assets/35aba87c-f222-4841-8adf-b18d0f600971)

It would actually look to be correct, going from what @ewoudwerkman states. The supply pipe runs in North-Western direction, the return pipe in South-Eastern direction, so the first coordinate of the supply pipe LineString geometry is at the joint, not the demand point.

Or should supply and return pipe geometries follow the same direction in ESDL?

**Comment by WouterSpaak** [comment]
Note that ports going *into* return pipes are always rendered correctly.

**Comment by ewoudwerkman** [comment]
If I look at the ESDL and specifically look at the order of de ports, I see that the list starts with an `OutPort` and then an `InPort`. If you flip these, the visualisation is Ok again:

![image](https://github.com/user-attachments/assets/427a7b66-68eb-4ad7-ac5c-3d55593007b7)


which means the first point of the geometry refers to the first port (the `OutPort`) `lat="52.06899416979089" lon="5.118928129649979"` (top-left) and the last point `lat="52.06805057736153" lon="5.123006535837114"` to the 2nd port (the `InPort`, bottom-right).
The `OutPort` (top-left) of Pipe 4 is connected to Joint 3 (located bottom-right) and the `InPort` (bottom-right) is connected to Joint 2 (located top-left), hence the connections that cross.

**Comment by WouterSpaak** [comment]
Oh wow, so the *order* of declarations in the XML DOM matters? That seems a bit smelly. But also a wildly easy fix for us: we build the XML document on the fly from PostGIS features, so I just need to swap two items in an array. Will update this comment thread tomorrow to see if that does the trick. If so, I'll post a screenshot and resolve this issue. :)

**Comment by ewoudwerkman** [comment]
Yes, the order of items in lists in ESDL matter (and also for XML or HTML in general, thankfully). For example the list of points in a geometry also matter, otherwise e.g. lines and polygons are rendered wrongly. Internally we use ordered lists, such that the order is always correct. 

I saw in the example XML that some assets have the order [InPort, OutPort] and some have [OutPort, InPort] in their port list, so please check what the correct order should be.
